tests/test_mail.py

```diff
@@ -108,9 +108,8 @@ def test_send_attach_utf8(self):
 
         msg = self.catched_msg['msg']
         self.assertEqual(msg['subject'], subject)
-        self.assertEqual(msg.get_charset(), Charset('utf-8'))
-        self.assertEqual(msg.get('Content-Type'),
-                         'multipart/mixed; charset="utf-8"')
+        self.assertEqual(msg.get_charset(), None)
+        self.assertEqual(msg.get('Content-Type'), 'multipart/mixed')
```

Comment:
Isn’t this a refression? Shouldn’t the old test expectations continue to work? Otherwise, maybe we are doing something wrong.

Reply:
Thanks @Gallaecio , yes, I agree, that's why I set old tests as were before and added a new similar one. The implemented approach seems the only compatible way to avoid regressions and fix the issue. 
The main difference I can see is that if you try to revert changes on test and instead of testing:
self.assertEqual(msg.get_payload(decode=True).decode(), body) -> self.assertEqual(msg.get_payload, body.encode('utf-8')) make tests not failed, because payload attribute is overwritten at the end of set_charset() by this line:
 self._payload = charset.body_encode(payload), but maybe that's ok.
I made also a test with two attachments to test.